### PR TITLE
refactor(bdk_hwi): remove `bdk_hwi`, as `HWISigner`'s being moved to `rust-hwi`

### LINKED ISSUE #1516: Move `bdk_hwi` to `hwi` module
**Describe the enhancement**  

We should move `bdk_hwi` from the `bdk` repo crates and include it with to the `hwi` crate. 

**Use case**  

Removing `bdk_hwi` will simplify the `bdk` workspace and ci testing because we won't need to use pyo3.

**Additional context**

If we determine no one is using the `hwi` module we can more easily deprecate both `hwi` and `bdk_hwi` when they're together in the same crate. But also they will be together in case someone does want to use and maintain them.

### PR DESCRIPTION
**Pull request by oleonardolima**
fixes #1516
depends on https://github.com/bitcoindevkit/rust-hwi/pull/104

### Description

This PR removed the `bdk_hwi` crate, and the dangling `hardwaresigner.rs` file from `bdk_wallet` crate.

It simplifies the BDK project workspace as it removes the dependency on `rust-hwi`, pyo3 and some required steps on CI.

The removal is fine, as the `HWISigner` is being moved to `rust-hwi` project instead, please check: https://github.com/bitcoindevkit/rust-hwi/pull/104

### Notes to the reviewers

Please let me know what you think about the removal, and if I missed anything.

### Changelog notice

- Removed dangling unused `hardwaresigner.rs` file from `bdk_wallet` crate.
- Removed `bdk_hwi` crate, moved to `rust-hwi` project.

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [ ] I've added tests for the new feature
* [ ] I've added docs for the new feature

#### Bugfixes:

* [ ] This pull request breaks the existing API
* [ ] I've added tests to reproduce the issue which are now passing
* [x] I'm linking the issue being fixed by this PR

### CODE REVIEW

File: crates/hwi/src/signer.rs [hunk 1]
```diff
@@ -1,94 +0,0 @@
...
-//
-//         let addr = wallet.get_address(bdk_wallet::AddressIndex::LastUnused);
-//         let mut builder = wallet.build_tx();
-//         builder.drain_to(addr.script_pubkey()).drain_wallet();
-//         let (mut psbt, _) = builder.finish().unwrap();
-//
-//         let finalized = wallet.sign(&mut psbt, Default::default()).unwrap();
-//         assert!(finalized);
-//     }
-// }
```
Comment (oleonardolima) [inline crates/hwi/src/signer.rs:94]: I don't think this a test we can move, at least not in its current state as it depends on `bdk_wallet::common::tests` module.

I'll need to rework it on the new `hwi::signer` to cover what some helper fns are doing manually.

**Comment by notmandatory** [comment]
bitcoindevkit/rust-hwi#104 is merged.

**Comment by notmandatory** [comment]
I rebased this PR and removed the `hwi/Cargo.toml` file which looks like it was missed.

**Comment by oleonardolima** [comment]
> I rebased this PR and removed the `hwi/Cargo.toml` file which looks like it was missed.

Thanks! I just noticed that I forgot to remove the `automation.json` file used by the simulator, I removed it in cd8ddfe18441c562342bcaccf7cae3a0d9ebbd92

**Review by notmandatory — APPROVED** [review]
ACK cd8ddfe18441c562342bcaccf7cae3a0d9ebbd92

**Pull request merged**